scripts/color-sorting.js

```diff
@@ -0,0 +1,101 @@
+/**
+ * @fileoverview
+ * Color sorting algorithm.
+ */
+
+module.exports = {
+  sortByColors: function (colors) {
+    const isGray = (rgb, range) => {
+      var { r, g, b } = { ...rgb };
+      return (
+        r >= g - range &&
+        r <= g + range &&
+        b >= g - range &&
+        b <= g + range &&
+        r >= b - range &&
+        r <= b + range
+      );
+    };
+
+    const rgbToHsl = (rgb) => {
+      var { r, g, b } = { ...rgb };
+      // Normalize r, g, and b
+      r /= 255;
+      g /= 255;
+      b /= 255;
+
+      // Find greatest and smallest channel values
+      let cmin = Math.min(r, g, b),
+        cmax = Math.max(r, g, b),
+        delta = cmax - cmin,
+        h = 0,
+        s = 0,
+        l = 0;
+
+      // Calculate hue
+      // No difference
+      if (delta == 0) h = 0;
+      // Red is max
+      else if (cmax == r) h = ((g - b) / delta) % 6;
+      // Green is max
+      else if (cmax == g) h = (b - r) / delta + 2;
+      // Blue is max
+      else h = (r - g) / delta + 4;
+
+      h = Math.round(h * 60);
+
+      // Ensure all hues are positive
+      if (h < 0) h += 360;
+      // Calculate luminosity
+      l = (cmax + cmin) / 2;
+
+      // Calculate saturation
+      s = delta == 0 ? 0 : delta / (1 - Math.abs(2 * l - 1));
+
+      // Multiply l and s by 100 to get the value in percent, rather than [0,1]
+      s = +(s * 100);
+      l = +(l * 100);
+      return { h, s, l };
+    };
+
+    const GREY_RANGE = 10;
+    const BLACK_CUTOFF = 15;
+    const WHITE_CUTOFF = 80;
```

Comment:
The old website didn't have a cutoff point for whites, and to be honest I don't feel like we need it - at least with the current set of colors. I definitely think `80` is too low and it's putting some colors with the whites that I wouldn't expect to be there. I would be fine with setting it to either 90 (for symmetry) or anything above that, which currently doesn't put any color in this bucket.

```suggestion
    const WHITE_CUTOFF = 90;
```

Reply:
we need it to remove the whites from general ordering of color icons.
We could put the threshold to 90, but be warned that while that will remove the light magenta and light pink, it would also leave a lot of white colors sprinkle throughout the general color array.